opennms-web-api/src/main/java/org/opennms/web/svclayer/support/DefaultAdminCategoryService.java

```diff
@@ -487,6 +487,10 @@ public EditModel(OnmsCategory category, List<OnmsNode> nodes,
             for (OnmsNode node : memberNodes) {
                 m_nodes.remove(node);
             }
+                
+            m_nodes =
+                    new ArrayList<OnmsNode>(nodes);
```

Comment:
Reconstituting m_nodes this will cause the list of available nodes (the left-hand list) to contain all nodes, including ones already in the category. I think all you really need to do is the Collections.sort(m_nodes).